### PR TITLE
feat: Add helpers to create nullable properties from `Option[A]`

### PR DESCRIPTION
**Pull request by OddKristensen**
It struck me that if `Json.fromString("foo")` is a quick and easy way to construct the JSON property for the string, then it should be just as easy to do that for an `Option[String]`. So I added a bunch of helpers that account for this.

You can argue that this generalizes to be:
```scala
final def propertyOrNull[A](value: Option[A])(f: A => Json): Json = value.fold(Json.Null)(f)
```

I did consider adding this generalization, but seemed a little too general perhaps.

### CODE REVIEW

File: modules/tests/shared/src/test/scala/io/circe/JsonSuite.scala [hunk 1]
```diff
@@ -269,6 +269,30 @@ class JsonSuite extends CirceMunitSuite with FloatJsonTests {
     assert(result === expected && resultAlias === expected)
   }
 
+  test("fromStringOrNull return Null on None") {
+    assertEquals(Json.fromStringOrNull(None), Json.Null)
+  }
```
Comment (zarthross) [inline modules/tests/shared/src/test/scala/io/circe/JsonSuite.scala:274]: Can you also add tests for the `Some` side here?  As it stands now we could accidentally always return null from `from*OrNull` and the tests would pass.

**Review by zarthross — DISMISSED** [review]

**Review by zarthross — APPROVED** [review]
